### ISSUE TITLE
[FR] Home screen refresh

### ISSUE THREAD

**Issue by vdatanet** [jellyfin/Swiftfin#123]
It would be cool if the home screen was refreshed when bringing back the application from background to foreground and to have the swipe down action to refresh the home screen.

Thanks!

**Comment by winterqt** [comment]
This also applies to when navigating back from the video player, as currently you have to force close and reopen the app for any of the screens to refresh. (another example being the item detail page)

**Comment by LePips** [comment]
I looked at work for this and I think we should wait until iOS 15 is released because they made is easy and clean.

So, might be a few weeks. I can make a draft PR soon with the work and have it wait until iOS 15 launches.

**Comment by winterqt** [comment]
Sounds good, I'll be sure to test it once you PR.

Out of curiosity, what API are you speaking of?

**Comment by LePips** [comment]
https://developer.apple.com/documentation/swiftui/view/refreshable(action:)?language=_2

**Comment by LePips** [comment]
Bummer, didn't read that it only supports `List` because it's actually a `UITableView` underneath. We use a `ScrollView` with an embedded `VStack`. Making a refreshable `ScrollView` isn't too difficult (at least looking at it) so this means:
1 - We will have to create a `RefreshableScrollView` which there are tutorials online for
2 - We won't be able to use nice SwiftUI features, but we'll live

So, now the wait is purely on implementation instead of waiting for iOS 15

**Comment by PangMo5** [comment]
First, It looks good to replace all existing scrolling VStack with LazyVStack.

And I use the source in my personal project.
https://github.com/siteline/SwiftUIRefresh/blob/master/Sources/PullToRefresh.swift
I don't know if this code works with LazyVStack, not List.

Additionally, changing existing LazyVStack to List may result in unwanted UI such as a separator.

**Comment by LePips** [comment]
We can customize the list styles, removing the edge insets and separators. `List` is also lazy loaded. 

The `VStack` isn't the problem with pull to refresh, it's the `ScrollView`. From just glancing at what you use, it looks for a `UITableView` in the view hierarchy, which means it has to be used in a `List`. 

I tried some pre-made ones online that were trying to work with `ScrollView` but the way we have our views set up they wont work for us. So, it would be better now and in the future to switch to `List`, which would also allow us to use `.refreshable()`.

**Comment by LePips** [comment]
Chewing on my own words here and this has become more complicated than it's probably worth. We use a stretchy header for the media which is in a `ScrollView`. I have looked online a ton to find a solution for stretchy headers for `List` and there are some for `UITableView` but we can't use that. Any pull to refresh solution that I've found for `ScrollView` doesn't work well with the large navigation title that's used on the home screen, but may work on the media views.

Essentially, SwiftUI isn't powerful enough yet. I'm thinking of just having an explicit refresh button for the home screen somewhere (probably in Settings to refresh watched content and sync with the server) because I'm _thinking_ that people don't need to sync with their server that often and making it a more explicit action is okay. For the media views, I have some other ideas for how metadata can be refreshed (not pull to refresh).

I'll close this issue soon if nobody else has anything to comment.

**Comment by winterqt** [comment]
The only comment I have is that I think the home screen and media item views should update like they do in the web / other clients: if you return to them after watching something, it'll update your progress / watched state, which currently requires restarting the app.

**Comment by LePips** [comment]
Yeah that can be a part of that, we'll do a full refresh of metadata and other statistics.
